core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
What our investigation found, is that, _with a newly created topic_, the isr-expiration thread may call `maybeShrinkIsr` before the lastCaughtUpTime gets updated, when its value is still -1, and therefore shrinks the ISR and the metric UnderReplicatedPartitionCount goes up - temporarily.

This is demonstrated by our unit test, if you remove 
`filter(r => !r.ignoreIsrCheck)`
the test will fail. QED.

The `ignoreIsrCheck` allows to ignore this check to be ignored on a new Replica for one period of the ISR scheduled thread, i.e. `period = config.replicaLagTimeMaxMs / 2`
We think that is acceptable.